### ISSUE TITLE
Error: Unable to find Rstudio Rproj file at the root of the workflowr project. Did you delete it?

### ISSUE THREAD

**Issue by BadSkittles** [workflowr/workflowr#61]
Hi,

I am interested in using workflowr in the future, but I am running into some issues early on. I am currently following the guide https://jdblischak.github.io/workflowr/articles/wflow-01-getting-started.html posted by John Blischak. So, I have created a folder on the desktop, and a r studio markdown file saved in the newly created folder. This is the code for it, and everything works perfectly fine at this point. 

```r{}
library("workflowr")
wflow_start("myproject")
wflow_build()
wflow_view()
```

However, once I change the code in the r markdown file to simply this 

```{r}
library("workflowr")
wflow_build()
wflow_view()
```
and run the markdown file for the second time, I receive the error, "Error: unable to find RStudio RProj file at the root of the workflowr directory. Did you delete it?" The thing is that I never even touched the workflowr directory. I tried to diagnose the problem by downloading the files in you guys' git hub repository at https://github.com/jdblischak/workflowr. Once I manually added the RStudio RProj file into the workflowr directory, I tried to execute the above code once more. I get the error message that _site.yml is not in the analysis folder. However, I checked inside the analysis folder, and I see that _site.yml is in the folder. Do you guys have any ideas of what might be going on?? Sorry for the long winded question.

**Comment by pcarbo** [comment]
@BadSkittles Thanks for sharing. I'm not sure if adding workflowr commands within an R Markdown document is a good idea---it is a bit circular, and I could see that it could create unforeseen problems. Can you run `getwd()`? It is possible that you are in the wrong working directory, which may be the root of your problems.

@jdblischak Also, the error could be more informative---maybe something like, "Error: unable to find RStudio RProj file at the root of the workflowr directory. Did you delete it, or forget to change your working directory to the workflowr project directory?"

**Comment by jdblischak** [comment]
@BadSkittles Thanks for your interest in using workflowr and for the detailed explanation of your issue.

To start I'll restate what @pcarbo advised. The workflowr package supplies function for creating and managing a version controlled website from R Markdown files. The R Markdown files should contain your analyses and any other content you wish to host on your website. The workflowr functions themselves should be run directly in the R console. I'll update the documentation to better convey this.

Thus the quick solution to your issue is that you should let workflowr create the folder on your Desktop instead of doing it yourself manually. Please try these steps:

1. Delete the folder you had previously created on your Desktop.
2. Open RStudio.
3. In the R console, run the following lines. This will both create the new folder with all the workflowr files and also change the current working directory to this new folder.
    ```
   library("workflowr")
   wflow_start("~/Desktop/myproject")
    ```
4. Continue following the [Getting Started vignette](https://jdblischak.github.io/workflowr/articles/wflow-01-getting-started.html)

Now for the more detailed explanation of what was happening from what you described. When you put this code below into an R Markdown file and click the RStudio button Knit HTML, it creates the workflowr project "myproject", switches to that directory, builds it, and then views it.

```
library("workflowr")
wflow_start("myproject")
wflow_build()
wflow_view()
```

However, when you put the following in the R Markdown file and run it, you get an error because the working directory is the directory you had created, not the workflowr project. Thus you get an error from `wflow_build()` about now being able to find an RStudio project file.

```
library("workflowr")
wflow_build()
wflow_view()
```

Please let me know if you have any questions about this explanation.

**Comment by jdblischak** [comment]
> Also, the error could be more informative---maybe something like, "Error: unable to find RStudio RProj file at the root of the workflowr directory. Did you delete it, or forget to change your working directory to the workflowr project directory?"

@pcarbo Agreed. The current error message doesn't include the other potential reason for the error, not being in the workflowr directory, which was the issue in this instance.

**Comment by jdblischak** [comment]
@BadSkittles @pcarbo I've updated the error message and added some extra text at the beginning of the Getting Started vignette in commit https://github.com/jdblischak/workflowr/commit/439821d8a1d054509bd930c0eaf7651152d9658b Please let me know if you have any feedback (also note that these changes won't be live on the online documentation until the next release).

**Comment by pcarbo** [comment]
@jdblischak Looks good!

**Comment by jdblischak** [comment]
@BadSkittles Please don't hesitate to ask follow up questions here. I am interested not only in helping you get setup with workflowr, but also improving the documentation based on your feedback.

**Comment by BadSkittles** [comment]
Thank you for the incredible support. Yes, you guys were right! It was simply a work directory issue, and I shouldn't have ran the commands through a r markdown document/manually created a folder. Also, the error message looks good to me!

One thing I noticed is that I invoke csv files and png images in my previous r markdown documents, and the data become broken once i transport these Rmd files to the analysis folder. Afterwards, I would have to transport my previous csv and images into the analysis folder as well, but then the folder gets very clustered. Is there a feature in the package where I can simply store all my images or csv files separate from the analysis Rmds and then have the analysis Rmds find them there?

**Comment by pcarbo** [comment]
@BadSkittles I believe you need to put your images in the `docs` folder (or subfolders, e.g., `docs/images`. See [this repo](https://github.com/stephenslab/wflow-divvy) for an example of how to organize and refer to your data sets in the R Markdown files.

**Comment by jdblischak** [comment]
> Is there a feature in the package where I can simply store all my images or csv files separate from the analysis Rmds and then have the analysis Rmds find them there?

@BadSkittles Migrating an existing project to use the workflowr framework takes a little bit of work. To start, I would follow these [directions from the vignette on migrating projects](https://jdblischak.github.io/workflowr/articles/wflow-03-migrating.html#scenario-i-have-a-collection-of-r-markdown-files).

By running `wflow_update()` after moving your R Markdown files to `analysis/`, all your R Markdown files will be using the workflowr template. The template settings automatically save the figure files created by the R Markdown files to specific locations in order to commit them to Git.

For the data files, I'd recommend moving them to the subdirectory `data/`. Then in your R Markdown files, change the paths to the files from, e.g. `"file.csv"` to `../data/file.csv`.

Do you have external images that you want to include in your site that are not created automatically when building the R Markdown files? This is what @pcarbo was providing instructions for. If yes, you could move the png files into `docs/images/` and then refer to them in the R Markdown files using the Markdown syntax: `![](images/file.png)`. You omit `docs/` in the path to the image file because the path needs to be relative to the HTML file in `docs/`. A similar suggestion was made here: https://github.com/jdblischak/workflowr/issues/37#issuecomment-294934805

**Comment by BadSkittles** [comment]
Cool, storing images in a docs folder worked for me. I've also noticed that Latex code doesn't seem to work in the r markdown files inside the analysis folder. Is that the case?

**Comment by pcarbo** [comment]
@BadSkittles You mean for math equations? [Here](https://stephenslab.github.io/wflow-divvy/time-of-day-trends.html) is an example using `workflowr` with equations (you need to scroll down a bit). Do you see them in your browser?

**Comment by BadSkittles** [comment]
So, I would often write code such as this in my r markdown document, which just prints out helloooo the way that I prefer it to. 

\newpage
\begin{table}[htp]
\centering
\fontsize{40}{44}
\fontseries{b}
\selectfont
Helloooo
\end{table}

However, it doesn't work in the R markdown files that I have in the analysis folder. But these markdown files are ordinarily exported as PDFs instead of html. Maybe that's why this line of code doesn't work?

**Comment by pcarbo** [comment]
@BadSkittles You can only insert Latex into R Markdown documents that generate a PDF. See [here](http://rmarkdown.rstudio.com/pdf_document_format.html) for details.

**Comment by jdblischak** [comment]
> Cool, storing images in a docs folder worked for me.

@BadSkittles Great!

> I've also noticed that Latex code doesn't seem to work in the r markdown files inside the analysis folder. Is that the case?

You are bound by the capabilities of [pandoc](http://pandoc.org/). workflowr calls [rmarkdown](http://rmarkdown.rstudio.com) which calls [pandoc](http://pandoc.org/). In theory an R Markdown document can be converted to HTML, PDF, and Word, but in practice this is only true if the file has not been heavily specialized for one format. Once you start to use LaTeX macros to customize the appearance of the PDF or HTML tags to customize the appearance of the HTML, at best the formatting won't translate to the other formats (and at worst, it will cause an error). The link that @pcarbo provided is a good place to learn more.

In general, if your R Markdown file can be built with the code below, then it should be fine to include in a workflowr project.

```
library("rmarkdown")
render("file.Rmd", html_document())
```

**Comment by jdblischak** [comment]
Closing this Issue now that the improved error message when workflowr functions are run outside of a workflowr project has been merged into master in release 0.8.0.

@BadSkittles If you have future issues with R Markdown files, please feel free to open a new issue.
